kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorIT.java

```diff
@@ -0,0 +1,472 @@
+        .put("myBool", true)
+        .put("myInt32", 32)
+        .put("myFloat32", 3.2f)
+        .put("myFloat64", 64.64)
+        .put("myString", "theStringVal");
+  }
+
+  /**
+   * Get an S3 client based on existing credentials, or a mock client if running on jenkins.
+   *
+   * @return an authenticated S3 client
+   */
+  private static AmazonS3 getS3Client() {
+    if (useMockClient()) {
+      S3Mock api = new S3Mock.Builder().withPort(MOCK_S3_PORT).withInMemoryBackend().build();
+      api.start();
+      /*
+       * AWS S3 client setup.
+       * withPathStyleAccessEnabled(true) trick is required to overcome S3 default
+       * DNS-based bucket access scheme
+       * resulting in attempts to connect to addresses like "bucketname.localhost"
+       * which requires specific DNS setup.
+       */
+      EndpointConfiguration endpoint = new EndpointConfiguration(MOCK_S3_URL, AWS_REGION);
+      AmazonS3 mockClient = AmazonS3ClientBuilder
+          .standard()
+          .withPathStyleAccessEnabled(true)
+          .withEndpointConfiguration(endpoint)
+          .withCredentials(new AWSStaticCredentialsProvider(new AnonymousAWSCredentials()))
+          .build();
+
+      log.info("No credentials found, using mock S3 client.");
+      return mockClient;
+    } else {
+      log.info("Credentials found, using real S3 client.");
+      // DefaultAWSCredentialsProviderChain,
+      // assumes .aws/credentials is setup and test bucket exists
+      return AmazonS3ClientBuilder.standard().withRegion(AWS_REGION).build();
+    }
+  }
+
+  /**
+   * Clear the given S3 bucket. Removes the contents, keeps the bucket.
+   *
+   * @param bucketName the name of the bucket to clear.
+   */
+  private static void clearBucket(String bucketName) {
+    for (S3ObjectSummary file : S3Client.listObjectsV2(bucketName).getObjectSummaries()) {
+      S3Client.deleteObject(bucketName, file.getKey());
+    }
+  }
+
+  /**
+   * Check if the file names in the bucket have the expected namings.
+   *
+   * @param bucketName        the name of the bucket with the files
+   * @param expectedTopic     the expected topic in the file path and file name
+   * @param expectedPartition the expected partition number in the file path and name
+   * @param expectedExtension the expected extensions of the files
+   * @return whether all the files in the bucket match the expected values
+   */
+  private boolean fileNamesValid(String bucketName, String expectedTopic, int expectedPartition,
+      String expectedExtension) {
+    for (S3ObjectSummary file : S3Client.listObjectsV2(bucketName).getObjectSummaries()) {
+      S3FileInfo fileInfo = new S3FileInfo(file.getKey());
+      if (!fileInfo.namingIsValid()
+          || !fileInfo.filenameTopic.equals(expectedTopic)
+          || fileInfo.directoryPartition != expectedPartition
+          || !fileInfo.extension.equals(expectedExtension)
+      ) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
probably should just generate a list of expected files depending on the flush size and partitioner